lib/Dialect/FIRRTL/Transforms/CheckInit.cpp

```diff
@@ -0,0 +1,189 @@
+//===- CheckInit.cpp - Ensure all wires are initialized ------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines the CheckInit pass.  This pass checks that all wires are
+// initialized (connected too).
+//
+//===----------------------------------------------------------------------===//
+
+#include "PassDetails.h"
+#include "circt/Dialect/FIRRTL/FIRRTLFieldSource.h"
+#include "llvm/Support/Debug.h"
+
+#define DEBUG_TYPE "firrtl-check-init"
+
+using llvm::BitVector;
+
+using namespace mlir;
+using namespace circt;
+using namespace firrtl;
+
+static void clearUnder(BitVector &bits, Type t, ArrayRef<int64_t> path,
+                       uint64_t fieldBase = 0) {
+  if (auto bundle = dyn_cast<BundleType>(t)) {
```

Comment:
`FIRRTLTTypeSwitch` , or if prefer it this way probably should use `firrtl::type_dyn_cast`